### PR TITLE
sink(ticdc): add knob to controll enable kafka sink v2 and add basic dependency scaffold

### LINKED ISSUE #8081: add knobs to controller enable kafka sink v2

### LINKED ISSUE #8082: implement kafka-go manual balancer to allow set partition directly.

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve?

Issue Number: close #8081, close #8082 

### What is changed and how it works?

* add `enable-kafka-sink-v2` to the `Debug` configuration.
* update go mod to import kafka-go
* add a new folder `pkg/sink/kafka/v2`and `muanal_balancer.go` as the placeholder.

#### Tests 

 - Unit test

### Release note 

```release-note
`None`
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -295,3 +296,6 @@ replace github.com/benbjohnson/clock v1.3.0 => github.com/benbjohnson/clock v1.1
 
 // copy from TiDB
 replace go.opencensus.io => go.opencensus.io v0.23.1-0.20220331163232-052120675fac
+
+// todo: remove this replace after https://github.com/segmentio/kafka-go/pull/1059 merged
+replace github.com/segmentio/kafka-go v0.4.38 => github.com/3AceShowHand/kafka-go v0.0.0-20230109084533-cd584ac8c705
```
Comment (overvenus) [inline go.mod:301]: Do we have to rely on the PR?
What about using `github.com/pingcap/kafka-go`?

**Review by overvenus — COMMENTED** [review]
Rest LGTM

File: go.mod (hunk 1 above)
Comment (3AceShowHand) [reply go.mod:301]: > Do we have to rely on the PR? What about using `github.com/pingcap/kafka-go`?

The master branch of the kafka-go does not support our desired feature (allow customizable field) at the moment, so we decide to use our development branch temporarily.

**Review by sdojjy — APPROVED** [review]

**Review by overvenus — APPROVED** [review]

**Pull request merged**